packages/x-components/src/x-modules/search-box/components/__tests__/search-input-placeholder.spec.ts

```diff
@@ -0,0 +1,188 @@
+import { mount, Wrapper } from '@vue/test-utils';
+
+import { getDataTestSelector, installNewXPlugin } from '../../../../__tests__/utils';
+import { getXComponentXModuleName, isXComponent } from '../../../../components';
+import { XPlugin } from '../../../../plugins';
+import { XEvent } from '../../../../wiring';
+import { searchBoxXModule } from '../../x-module';
+import SearchInput from '../search-input.vue';
+import SearchInputPlaceholder from '../search-input-placeholder.vue';
+
+async function mountTestComponent({
+  messages = ['Find shirts', 'Find shoes', 'Find watches', 'Find handbags', 'Find sunglasses'],
+  animationIntervalMs,
+  animateOnlyOnHover,
+  autofocus = false
+}: MountTestComponentOptions = {}): Promise<MountTestComponentAPI> {
+  const [, localVue] = installNewXPlugin();
+  XPlugin.registerXModule(searchBoxXModule);
+
+  const parent = document.createElement('div');
+  document.body.appendChild(parent);
+
+  const wrapper = mount(
+    {
+      template: `
+        <div>
+          <SearchInputPlaceholder
+            :messages="messages"
+            :animationIntervalMs="animationIntervalMs"
+            :animateOnlyOnHover="animateOnlyOnHover"
+          />
+          <SearchInput :autofocus="autofocus" />
+        </div>
+      `,
+      props: ['messages', 'animationIntervalMs', 'animateOnlyOnHover', 'autofocus'],
+      components: {
+        SearchInputPlaceholder,
+        SearchInput
+      }
+    },
+    {
+      localVue,
+      /*
+       * In order to make the autofocus test work after the jest 27 update, now is mandatory to
+       * attach the element to some parent in the DOM, to emit the focus event.
+       */
+      attachTo: parent,
+      propsData: {
+        messages,
+        animationIntervalMs,
+        animateOnlyOnHover,
+        autofocus
+      }
+    }
+  );
+  await wrapper.vm.$nextTick();
+
+  function getPlaceholderText(): string {
+    return wrapper.find(getDataTestSelector('search-input-placeholder')).text();
+  }
+
+  function isPlaceholderVisible(): boolean {
+    return wrapper.find(getDataTestSelector('search-input-placeholder')).exists();
+  }
+
+  function hoverInput(mode: 'in' | 'out'): Promise<void> {
+    const inputWrapper = wrapper.findComponent(SearchInput);
+    const hoverEvent = {
+      in: 'mouseenter',
+      out: 'mouseleave'
+    }[mode];
+    inputWrapper.trigger(hoverEvent);
+    return wrapper.vm.$nextTick();
+  }
+
+  async function emit(event: XEvent, payload: any): Promise<void> {
+    wrapper.vm.$x.emit(event, payload);
+    await localVue.nextTick();
+  }
+
+  return {
+    messages,
+    wrapper,
+    getPlaceholderText,
+    isPlaceholderVisible,
+    hoverInput,
+    emit
+  };
+}
+
+interface MountTestComponentOptions {
+  messages?: Array<string>;
```

Comment:
Comment these properties

Reply:
document*

Reply:
Also, Move the interfaces below the describe, at the end of the file